### PR TITLE
Core fixes and iterable additions

### PR DESCRIPTION
**Pull request by JonasWanke**
### Checklist

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: packages/Core/iterable.candy [hunk 1]
```diff
@@ -31,10 +32,10 @@ isEmpty iterable :=
 generateWithStateHelper = { generateWithStateHelper state mapper ->
   {
     mapper state | result.map { result ->
-      [newItem, newState] = result
+      [item, state] = result
```
Comment (MarcelGarus) [inline packages/Core/iterable.candy]: Just noticing that we could also inline the entire function using the `recursive` function.

File: packages/Core/text.candy [hunk 2]
```diff
@@ -2,7 +2,7 @@ bool = use "..bool"
 [ifElse] = use "..controlFlow"
 [equals] = use "..equality"
 int = use "..int"
-iterable = use "..iterable"
+# iterable = use "..iterable"
```
Comment (MarcelGarus) [inline packages/Core/text.candy:5]: Why that?

**Review by MarcelGarus — CHANGES_REQUESTED** [review]

File: packages/Core/text.candy (hunk 2 above)
Comment (JonasWanke) [reply packages/Core/text.candy:5]: `iterable` imports `text`, so this causes a cycle. It was only used in the parameter validation of `fromUtf8`, which is currently commented out for performance reasons

File: packages/Core/iterable.candy (hunk 1 above)
Comment (JonasWanke) [reply packages/Core/iterable.candy]: We could also use that for the remaining helper functions in this file, but I'd wait for better pattern support (#295) before doing so

**Review by MarcelGarus — APPROVED** [review]

**Pull request merged**